packages/backend/src/peripherals/database/ReportRepository.ts

```diff
@@ -11,11 +11,45 @@ export interface ReportRecord {
   ethTVL: bigint
 }
 
+export interface ReportWithBalance extends ReportRecord {
+  balance: bigint
+}
+
 export class ReportRepository {
   constructor(private knex: Knex, private logger: Logger) {
     this.logger = this.logger.for(this)
   }
 
+  async getDaily(): Promise<ReportWithBalance[]> {
+    const rows = await this.knex('reports')
+      .select(
+        'reports.block_number',
+        'reports.unix_timestamp',
+        'reports.bridge_address',
+        'reports.asset_id',
+        'reports.usd_tvl',
+        'reports.eth_tvl',
+        'asset_balances.balance'
+      )
+      .leftJoin('asset_balances', function () {
+        this.on('asset_balances.block_number', '=', 'reports.block_number')
+          .andOn('asset_balances.asset_id', '=', 'reports.asset_id')
+          .andOn('asset_balances.holder_address', '=', 'reports.bridge_address')
+      })
+
+    return rows
+      .map((r) => ({
+        ...toRecord(r),
+        balance: r.balance ? BigInt(r.balance) : BigInt(0),
+      }))
+      .filter((r) => r.timestamp.isFull('day'))
+      .sort((a, b) => {
+        if (a.timestamp.lt(b.timestamp)) return -1
+        else if (a.timestamp.equals(b.timestamp)) return 0
+        else return 1
+      })
```

Comment:
1. You can filter in the sql query by selecting only timestamps that are divisible by 60 * 60 * 24 (number of seconds per day)
2. You can sort in the query by using order by